src/metrics-collector/db/sqldb/sqldb_test.go

```diff
@@ -0,0 +1,377 @@
+					BeforeEach(func() {
+						start = 234567
+						end = 555555
+					})
+
+					It("should not error and return correct metrics", func() {
+						Expect(err).To(BeNil())
+						Expect(len(mtrcs)).To(Equal(1))
+
+						instValues := []metrics.InstanceMetric{{123456, 0, "3333"}, {123476, 1, "6666"}}
+						Expect(mtrcs).To(ConsistOf(&metrics.Metric{
+							AppId:     "test-app-id",
+							Name:      metrics.MemoryMetricName,
+							Unit:      metrics.UnitBytes,
+							TimeStamp: 333333,
+							Instances: instValues,
+						}))
+					})
+				})
+			})
+		})
+	})
+
+	Describe("PruneMetrics", func() {
+		BeforeEach(func() {
+			db, err = NewSQLDB(conf, logger)
+			if err != nil {
+				Fail("can not connect database")
+			}
+			instances := []metrics.InstanceMetric{{123456, 0, "3333"}, {123476, 1, "6666"}}
+			metric := &metrics.Metric{
+				AppId:     appId,
+				Name:      metricName,
+				Unit:      metrics.UnitBytes,
+				TimeStamp: 222222,
+				Instances: instances,
+			}
+			db.SaveMetric(metric)
+
+			metric.TimeStamp = 333333
+			db.SaveMetric(metric)
+
+			metric.TimeStamp = 666666
+			db.SaveMetric(metric)
+
+		})
+
+		AfterEach(func() {
+			dropMetricsTable()
+		})
+
+		JustBeforeEach(func() {
+			err = db.PruneMetrics(before)
+		})
+
+		Context("when pruning metrics before all the timestamps of metrics", func() {
+			BeforeEach(func() {
+				before = 0
+			})
+
+			It("should not error and did not remove any metrics", func() {
+				Expect(err).To(BeNil())
+				Expect(getNumberOfMetrics()).To(Equal(3))
+			})
+		})
+
+		Context("when pruning all the metrics", func() {
+			BeforeEach(func() {
+				before = time.Now().UnixNano()
+			})
+
+			It("should not error and did not remove any metrics", func() {
+				Expect(err).To(BeNil())
+				Expect(getNumberOfMetrics()).To(Equal(0))
+			})
+		})
+
+		Context("when pruning part of the metrics", func() {
+			BeforeEach(func() {
+				before = 333333
+			})
+
+			It("should not error and did not remove any metrics", func() {
+				Expect(err).To(BeNil())
+				Expect(getNumberOfMetrics()).To(Equal(1))
+				Expect(hasMetric(appId, 666666)).To(BeTrue())
+			})
+		})
+
+	})
+
+	Describe("getAppIds", func() {
+		BeforeEach(func() {
+			db, err = NewSQLDB(conf, logger)
+			if err != nil {
+				Fail("can not connect database")
+			}
+			createPolicyTable()
+		})
+
+		AfterEach(func() {
+			dropPolicyTable()
+		})
+
+		JustBeforeEach(func() {
+			appIds, err = db.GetAppIds()
+		})
+
+		Context("when policy table is empty", func() {
+			It("should not error and return empty array", func() {
+				Expect(err).To(BeNil())
+				Expect(appIds).NotTo(BeNil())
+				Expect(len(appIds)).To(Equal(0))
+			})
+
+		})
+
+		Context("when policy table is not empty", func() {
+			BeforeEach(func() {
+				insertPolicy("first-app-id")
+				insertPolicy("second-app-id")
+				insertPolicy("third-app-id")
+			})
+
+			It("should not error and return all app ids", func() {
```

Comment:
It("returns all the app ids")
you can test for the error but you don't have to tell me in the test name.